blamo/lib/Export/CSVExporter.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:flutter/material.dart';
+import 'package:blamo/ObjectHandler.dart';
+import 'package:blamo/main.dart';
+import 'dart:io';
+import 'dart:async';
+import 'package:path_provider/path_provider.dart';
+import 'package:permission_handler/permission_handler.dart';
+
+class CSVExporter {
+  ObjectHandler objectHandler = new ObjectHandler();
+  StateData stateData;
+
+  LogInfo logInfo;
+  List<Unit> units;
+  List<Test> tests;
+  List<String> lines = [];
+
+  CSVExporter(this.stateData);
+
+  void exportToCSV() async{
+    String toWrite;
+    await getData();
+    buildCSVLines();
+    toWrite = formatLinesToString();
+    csvWrite(toWrite, stateData.currentDocument);
+  }
+
+  Future<int> getData() async{
+    logInfo = await objectHandler.getLogInfoData(stateData.currentDocument);
+    try{
+      units = await objectHandler.getUnitsData(stateData.unitList, stateData.currentDocument);
+    } catch (e){
+      debugPrint("(CSV)ERROR getting units");
+    }
+
+    try{
+      tests = await objectHandler.getTestsData(stateData.testList, stateData.currentDocument);
+    } catch (e){
+      debugPrint("(CSV)ERROR getting tests");
+    }
+    debugPrint("SUCCESS");
+    return 0;
+  }
+
+  void buildCSVLines(){
+    String header = "OBJECTID,Test Type,PROJECT,Number,Client,LATITUDE,LONGITUDE,North,East,Location,Elevation Datum,Borehole ID,Date Started,Date Completed,Surface Elevation (ft),Contractor/Driller,Method,Logged By,Checked By,Begin Depth (ft),End Depth (ft),Soil Type,Description,Moisture Content,Dry Density (pcf),Liquid Limit (%),Plastic Limit (%),Fines (%),Blows 1st,Blows 2nd,Blows 3rd,N Value";
+    lines.add(header);
+
+    //TODO SizeofTest
+    for(int i = 0; i < tests.length; i++){
+      lines.add("$i,"
+          "${tests[i].testType},"
+          "${logInfo.project},"
+          "${logInfo.number},"
+          "${logInfo.client},"
+          "${logInfo.lat},"
+          "${logInfo.long},"
+          ","
```

Comment:
Looks like you need to add another comma here

Reply:
Nah, I thought so too, but when you open it that offsets every field by 1. The example CSV I emailed out displays this current set up. with the extra comma it overflows the header fields.

Reply:
This problem was caused by commas in the text entry fields, should be fixed now.